linode/manager
# upcoming: [M3-8074] – Add "Disk Encryption" section to Linode Create flow

## Description 📝
Add "Disk Encryption" section to the Linode Create flow to allow user to choose if their VM will be encrypted or not.

## Changes  🔄
- `Disk Encryption` added as `AccountCapability` and region `Capabilities`
- New props added to `AccessPanel` component
- Conditional notice for Backups added in `AddonsPanel` component
- `useIsDiskEncryptionFeatureEnabled` hook added

## Target release date 🗓️
5/28/24

## Preview 📷
| Before (LDE not supported)  | After (LDE supported) |
| ------- | ------- |
| ![Screenshot 2024-05-16 at 5 42 04 PM](https://github.com/linode/manager/assets/114682940/a01a3171-7a33-442e-b5ce-324f71a33534) | ![Screenshot 2024-05-16 at 5 42 14 PM](https://github.com/linode/manager/assets/114682940/6fb575d3-4958-4462-a253-3fe9e6d310e7) |
| ![Screenshot 2024-05-16 at 5 44 11 PM](https://github.com/linode/manager/assets/114682940/165b8dc3-dcc4-400e-b5d9-89f157e902a7) | ![Screenshot 2024-05-16 at 5 45 04 PM](https://github.com/linode/manager/assets/114682940/46733e87-b3af-49cf-b6c2-60dc955da830) |
| ![Screenshot 2024-05-16 at 5 46 16 PM](https://github.com/linode/manager/assets/114682940/13b9bed4-604c-43d4-ac9a-be290c163a8f) | ![Screenshot 2024-05-16 at 5 46 42 PM](https://github.com/linode/manager/assets/114682940/25e6c4eb-4a96-4d42-b9a1-199b73bc8bae) |

## How to test 🧪
### Prerequisites
1. Have the `linode_disk_encryption` tag on your account
2. Point at the alpha/dev environment
3. Have the `Linode Disk Encryption (LDE)` flag in our dev tool toggled on

### Verification steps
Without the customer tag and/or with the LDE flag in our dev tool toggled off, you should not see any LDE-related things in the UI (or in the Linode Create payload). Otherwise,

1. "Disk Encryption" should be below "SSH Keys" in the same paper, and that paper should be labeled "Security"
 a. No region selected or region w/o Disk Encryption support selected --> "Encrypt Disk" checkbox should be unchecked and disabled with an explanatory tooltip; region w/ Disk Encryption support selected (Newark has support in alpha) --> checkbox should be enabled, and default to checked
 b.   Toggling the "Encrypt Disk" checkbox should work as expected
 c. The "Summary" section at the bottom should reflect your choice to encrypt or not, i.e., if "Encrypt Disk" is checked, you should see "Encrypted" in the summary
 d. "Encrypt Disk" checked & Backups add-on checked --> observe: notice under Backups stating, "Virtual Machine Backups are not encrypted."
3. `disk_encryption` shows with the expected value for cURL and Linode CLI when you click "Create Using Command Line" (and should be absent when the selected region doesn't support LDE)
4. No adverse or unexpected impacts where `<AccessPanel />` is used elsewhere (e.g., UDF fields, Rebuild from Image & Stackscript, Create Disk drawer --> Create from Image, etc.)

## As an Author I have considered 🤔
